test/performance/latency_test.go

```diff
@@ -77,50 +81,41 @@ func timeToServe(t *testing.T, img, query string, reqTimeout time.Duration) {
 		t.Fatalf("Error probing domain %s: %v", domain, err)
 	}
 
-	endpoint, err := spoof.ResolveEndpoint(clients.KubeClient.Kube, domain, test.ServingFlags.ResolvableDomain,
+	url, err := spoof.ResolveEndpoint(clients.KubeClient.Kube, domain, test.ServingFlags.ResolvableDomain,
 		pkgTest.Flags.IngressEndpoint)
 	if err != nil {
 		t.Fatalf("Cannot resolve service endpoint: %v", err)
 	}
 
-	opts := loadgenerator.GeneratorOptions{
-		Duration:       duration,
-		NumThreads:     1,
-		NumConnections: 5,
-		Domain:         domain,
-		URL:            fmt.Sprintf("%s/?%s", endpoint, query),
-		RequestTimeout: reqTimeout,
-		LoadFactors:    []float64{1},
-		FileNamePrefix: tName,
-	}
-	var flags []int
-	if !test.ServingFlags.ResolvableDomain {
-		flags = append(flags, loadgenerator.AddHostHeader)
-	}
-
-	resp, err := opts.RunLoadTest(flags...)
-	if err != nil {
-		t.Fatalf("Generating traffic via fortio failed: %v", err)
+	if !strings.HasPrefix(url, httpPrefix) {
```

Comment:
Seems like we should have a helper function for both sanitizing the URL and determining the headers to add, WDYT?

The tests would also have much less boilerplate as we could write:

```golang
targeter := vegeta.NewStaticTargeter(vegeta.Target{
  Headers: resolvedHeaders(test.ServingFlags.ResolvableDomain)
  URL: sanitizedURL(url) + ...
```

Maybe we should even have a wrapper that returns a `vegeta.Target` with all that stuff taken care of so we don't miss it like in the test below?

Reply:
Thanks for the suggestion! I have add `resolvedHeaders` and `sanitizedURL` as helper functions. I did not add the wrapper for building `vegeta.Target` as the function itself will be a bit verbose.